src/app/server/Dnssd.h

```diff
@@ -96,6 +96,12 @@ class DLL_EXPORT DnssdServer : public ICDStateObserver
 
     void SetICDManager(ICDManager * manager) { mICDManager = manager; };
 #endif
+
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+    template <class AdvertisingParams>
+    void AddTCPSupportToAdvertisement(AdvertisingParams & advParams);
```

Comment:
Why is this a template? why does this get wrapped instead of saying `advParams.SetTCPSupportModes(chip::Dnssd::TCPModeAdvertise::kTCPClientServer);`? It seems we are replacing one line of code with calling a template that implements one line of code. Why the extra indirection and complexity?

Reply:
Removed the function and replaced with a direct call to SetTCPSupportModes.